applicationset/generators/git.go

```diff
@@ -238,6 +258,40 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 	return res, nil
 }
 
+// filterAppsFiles filters a list of file paths based on inclusion and exclusion patterns defined in GitFileGeneratorItems.
+// It uses doublestar globbing to support complex patterns like "**/*.yaml".
+func (g *GitGenerator) filterAppsFiles(files []argoprojiov1alpha1.GitFileGeneratorItem, allFilePaths []string) []string {
+	res := []string{}
```

Comment:
```suggestion
	var res []string
```

Reply:
why would you prefer this?

Reply:
https://go.dev/wiki/CodeReviewComments#declaring-empty-slices

Reply:
It's a rather nit comment but `var res []string` is nil and technically uses less space.